webrender/src/device/gfx.rs

```diff
@@ -2170,6 +2187,116 @@ impl<B: hal::Backend> Device<B> {
             (device, queues.take(id).unwrap())
         };
 
+        #[cfg(feature = "headless_gfx")]
+        let (
+            surface_format,
+            depth_format,
+            render_pass,
+            framebuffers,
+            framebuffers_depth,
+            frame_depths,
+            frame_images,
+            viewport,
+            frame_count,
+        ) = {
+
+            let surface_format = ImageFormat::BGRA8;
+            let depth_format = hal::format::Format::D32Sfloat;
+            let render_pass = Device::create_render_passes(
+                &device,
+                hal::format::Format::Bgra8Unorm,
+                depth_format
+            );
+
+            let extent = hal::window::Extent2D {
+                width: window_size.0 as u32,
+                height: window_size.1 as u32,
+            };
+            let mut frame_depths = Vec::new();
+            let (frame_images, framebuffers, framebuffers_depth) = {
+                let extent = hal::image::Extent {
+                    width: extent.width as _,
+                    height: extent.height as _,
+                    depth: 1,
+                };
+                let mut images = Vec::new();
+                for _ in 0..2 {
+                    images.push(Image::new(
+                        &device,
+                        &memory_types,
+                        surface_format,
+                        extent.width as i32,
+                        extent.height as i32,
+                        1,
+                        hal::image::ViewKind::D2,
+                        1,
+                        hal::image::Usage::TRANSFER_SRC
+                            | hal::image::Usage::TRANSFER_DST
```

Comment:
out of curiosity, where do we use it as `TRANSFER_DST`?

Reply:
I think we need this bit set for `clear_image`: https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/vkCmdClearColorImage.html , and [here](https://github.com/szeged/webrender/blob/master/webrender/src/renderer.rs#L2293) we bind the default draw target and clear the frame image.